src/server.js

```diff
@@ -0,0 +1,38 @@
+const express = require("express");
+const { Resolver } = require("./resolver");
+
+class Server {
+  constructor() {
+    this.app = express();
+    this.resolver = new Resolver();
+    this.configure();
+  }
+
+  configure() {
+    this.app.get("/dnslink/:name", async (req, res) => {
+      const domainName = req.params.name;
+
+      try {
+        this.resolver.validateRequest(req);
+
+        const { skylink, sponsor } = await this.resolver.resolve(domainName);
+
+        res.json({ skylink, sponsor });
+      } catch (error) {
+        res.status(400).contentType("text/plain; charset=utf-8").send(error.message);
+      }
+    });
+  }
+
+  start(port, host) {
+    this.server = this.app.listen(port, (host = "0.0.0.0"), (error) => {
```

Comment:
I think it's more natural and a little bit less confusing to define a default argument value in the function declaration. In your code, `host` is undefined first and then the value changes when it's set inline so if we would add some code between, `host` would be undefined there.

```suggestion
  start(port, host = "0.0.0.0") {
    this.server = this.app.listen(port, host, (error) => {
```

Reply:
You're definitely right; that was my intention but I set it in a wrong place. As it is right now, setting `host` wouldn't even work, it would be overriden.